lang/java/reef-io/src/main/java/org/apache/reef/io/network/impl/NetworkConnectionFactory.java

```diff
@@ -73,16 +76,27 @@
     return connection;
   }
 
-  <T> Link<NetworkConnectionServiceMessage<T>> openLink(final Identifier remoteId) throws NetworkException {
+  Link<NetworkConnectionServiceMessage<T>> openLink(final Identifier remoteId) throws NetworkException {
     return networkService.openLink(remoteId);
   }
 
-  String getConnectionFactoryId() {
-    return this.connFactoryId;
+  @Override
+  public Identifier getConnectionFactoryId() {
+    return connectionFactoryId;
+  }
+
+  @Override
+  public Identifier getEndPointId() {
+    return endPointId;
   }
 
   Identifier getSrcId() {
-    return this.networkService.getNetworkConnectionServiceId();
+    // TODO : Change to always return the endPointId when the deprecated getNetworkConnectionServiceId is removed.
```

Comment:
I agree with deprecating current constructor and creating new one is better way, however, I think it can change some behavior of `NetworkConnectionService`. If we set the `endPointId` as `this.networkService.getNetworkConnectionServiceId()` in the constructor, `getSrcId()` only returns the identifier that was registered before the `NetworkConnectionFactory` created.

 If some existing users register another identifier to NetworkConnectionService after `NetworkConnectionFactory` was created, the sourceId of 'Message' is changed with the suggestion. (When `Message` is created by `Connection`, `getSrcId()` is called every time)